### PR TITLE
Add more words to the dictionary (compre, reviwer, supprot etc.)

### PR DESCRIPTION
**Pull request by qyearsley**

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -5406,8 +5407,9 @@ composit->composite
 compount->compound
 comppatible->compatible
 comprable->comparable
+compre->compare
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:5410]: Suggest compère too

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -19897,6 +19902,7 @@ quarentine->quarantine
 quartically->quadratically
 quatation->quotation
 quater->quarter
+quating->quoting
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:19905]: Suggest squatting

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -24118,6 +24128,8 @@ suppporting->supporting
 suppports->supports
 suppres->suppress
 suppressingd->suppressing
+supprot->support
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:24131]: Can you also add the s, ing, er and ers variants of this typo.

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -26908,6 +26920,8 @@ whicht->which
 whihc->which
 whihch->which
 whilest->whilst
+whiltelist->whitelist
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:26923]: Can you add the ing and ed variants of this typo.

File: codespell_lib/data/dictionary.txt [hunk 5]
```diff
@@ -21848,6 +21855,8 @@ reviewsectio->reviewsection
 revisisions->revisions
 revison->revision
 revisons->revisions
+reviwer->reviewer
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:21858]: Looking at my previous review in #1479 (I'm not quite sure why that got closed), here is another oustanding comment:
reviwed->reviewed and reviwing->reviewing

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (qyearsley) [reply codespell_lib/data/dictionary.txt:5410]: First time I've seen that word and slightly unusual in code, but OK

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (qyearsley) [reply codespell_lib/data/dictionary.txt:19905]: Also a lot less likely in code, but will do.

File: codespell_lib/data/dictionary.txt (hunk 5 above)
Comment (qyearsley) [reply codespell_lib/data/dictionary.txt:21858]: I closed it by mistake :-/ Thanks for reviewing again :-D

**Comment by qyearsley** [comment]
Thanks again peternewman :-)

File: codespell_lib/data/dictionary.txt (hunk 1 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt]: Done.

File: codespell_lib/data/dictionary.txt (hunk 2 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt]: Done.

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt:24133]: Done.

File: codespell_lib/data/dictionary.txt (hunk 4 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt:26929]: Done.

File: codespell_lib/data/dictionary.txt (hunk 5 above)
Comment (peternewman) [reply codespell_lib/data/dictionary.txt:21859]: Done.

**Comment by peternewman** [comment]
Thanks @qyearsley .

Two minor comments, if you can avoid force pushing if possible as it makes the review harder because you can't see what's changed since your last review.

Also I don't personally think we necessarily need to narrow the dictionary down to just programming (aside from the fact documentation can cover lots of broad subjects).

**Comment by qyearsley** [comment]
> Thanks @qyearsley .
> 
> Two minor comments, if you can avoid force pushing if possible as it makes the review harder because you can't see what's changed since your last review.

Thanks, will do next time! As you can tell I'm not yet very good at the GitHub PR review process :-P

**Pull request merged**